presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -123,8 +126,15 @@ public static ActualProperties deriveProperties(PlanNode node, List<ActualProper
         verify(node.getOutputSymbols().containsAll(localPropertyColumns), "Node-level local properties contain columns not present in node's output");
 
         // TODO: ideally this logic would be somehow moved to PlanSanityChecker
-        verify(node instanceof SemiJoinNode || inputProperties.stream().noneMatch(ActualProperties::isNullsReplicated) || output.isNullsReplicated(),
-                "SemiJoinNode is the only node that can strip null replication");
+        if (!(node instanceof SemiJoinNode)) {
+            Set<Replication> inputReplications = inputProperties.stream()
+                    .map(ActualProperties::getReplication)
+                    .filter(replication -> !replication.isNoReplication())
+                    .collect(toImmutableSet());
+            verify(inputReplications.size() == 0
+                            || (inputReplications.size() == 1 && Objects.equals(output.getReplication(), getOnlyElement(inputReplications))),
```

Comment:
But your code has different semantics than what was before. The closest of what was before is to check that all `inputReplications` are the same as `output.getReplication()`. There was no `inputReplications.size() == 1` condition before